shared/src/commonMain/kotlin/io/customer/shared/tracking/api/TrackingHttpClient.kt

```diff
@@ -0,0 +1,61 @@
+package io.customer.shared.tracking.api
+
+import io.customer.shared.tracking.api.model.BatchTrackingRequestBody
+import io.customer.shared.tracking.api.model.BatchTrackingResponse
+import io.customer.shared.tracking.api.model.BatchTrackingResponseBody
+import io.customer.shared.tracking.api.model.toTrackingRequest
+import io.customer.shared.tracking.model.Task
+import io.customer.shared.util.Logger
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+
+/**
+ * HTTP client interface responsible for making all network calls in the SDK.
+ *
+ * The class doesn't perform any validation and is only responsible for transforming and passing
+ * filtered data to the server.
+ */
+internal interface TrackingHttpClient {
+    /**
+     * Dispatch all tasks passed to the server in a single batch call.
+     */
+    suspend fun track(tasks: List<Task>): Result<BatchTrackingResponse>
+}
+
+internal class TrackingHttpClientImpl(
+    private val logger: Logger,
+    private val httpClient: HttpClient,
+) : TrackingHttpClient {
+    override suspend fun track(tasks: List<Task>): Result<BatchTrackingResponse> {
+        val result = kotlin.runCatching {
+            logger.debug("Batching ${tasks.size} track events")
+            val requestBody = BatchTrackingRequestBody(
+                batch = tasks.map { task ->
+                    // Ideally, tasks passed here should never have invalid identifier
+                    task.activity.toTrackingRequest(
+                        identityType = task.identityType,
+                        profileIdentifier = task.profileIdentifier.orEmpty(),
+                    )
+                },
+            )
+            val httpResponse = httpClient.post {
+                url { path(URLPaths.BATCH_TASKS) }
+                setBody(body = requestBody)
+            }
+            val responseBody = httpResponse.body<BatchTrackingResponseBody>()
+            val response = BatchTrackingResponse(
+                statusCode = httpResponse.status.value,
+                errors = responseBody.errors ?: emptyList(),
+            )
+            logger.debug("Batch tracking successful with code: ${response.statusCode}")
```

Comment:
Since we have a HTTP logger installed, I am not sure we need this logger line.

Reply:
Agreed, Removing this. Also, I don't think I have taken proper care of useful logs. So I have added this as an action item in pending tasks too.

Reply:
Thanks for sharing this. I think it is a good idea to start with. I've linked this comment with logs action.